src/nvim/ex_cmds2.c

```diff
@@ -3403,16 +3413,20 @@ static void script_host_execute_file(char *name, exarg_T *eap)
   // filename
   list_append_string(args, buffer, -1);
   // current range
-  list_append_number(args, eap->line1);
-  list_append_number(args, eap->line2);
+  assert(eap->line1 <= INT_MAX);
+  assert(eap->line2 <= INT_MAX);
+  list_append_number(args, (int)eap->line1);
+  list_append_number(args, (int)eap->line2);
```

Comment:
Makes sense, but if that is the case, what is the rationale behind having `linenr_T` as an alias for `long` instead of an `int`?

If we have a casting where code does not guarantee it is safe, I prefer to add an assertion. line numbers in any sane use case will fit in an int, but technically linenr_T is a long, so..

Apologies for being persistent, I'm trying to understand the rationale here.